packages/shared/src/hooks/useUpdateEmailNotificationSettings.ts

```diff
@@ -0,0 +1,246 @@
+      mutation useUpdateEmailNotificationSettingsMutation(
+        $enabledNotification: UpdateEmailNotificationSettingsInput!
+      ) @raw_response_type {
+        updateEmailNotificationSettings(input: $enabledNotification) {
+          ... on UpdateEmailNotificationSettingsPayload {
+            viewer {
+              email {
+                emailNotificationSettings {
+                  unsubscribedFromNotifications
+                  unsubscribedFromDigest
+                  unsubscribedFromMarketing
+                  unsubscribedFromMembersClub
+                  unsubscribedFromAll
+                }
+              }
+            }
+          }
+          ... on ErrInvalidInput {
+            __typename
+          }
+        }
+      }
+    `);
+
+  const updateEmailNotificationSettings = useCallback(
+    ({
+      unsubscribedFromNotifications,
+      unsubscribedFromDigest,
+      unsubscribedFromMarketing,
+      unsubscribedFromMembersClub,
+      unsubscribedFromAll,
+    }: updateEmailNotificationSettingsProps) => {
+      return updateEmailNotificationSettingsMutation({
+        variables: {
+          enabledNotification: {
+            unsubscribedFromNotifications,
+            unsubscribedFromDigest,
+            unsubscribedFromMarketing,
+            unsubscribedFromMembersClub,
+            unsubscribedFromAll,
+          },
+        },
+      });
+    },
+    [updateEmailNotificationSettingsMutation]
+  );
+
+  const handleEmailNotificationChange = useCallback(
+    async ({
+      settingType,
+      newSettingValue,
+      settingTitle,
+      pushToast,
+    }: handleEmailNotificationChangeProps) => {
+      const settingsUpdate = {
+        ...emailSettings,
+        [settingType]: newSettingValue,
+      };
+
+      try {
+        const response = await updateEmailNotificationSettings({
+          unsubscribedFromNotifications: !settingsUpdate.notifications,
+          unsubscribedFromDigest: !settingsUpdate.digest,
+          unsubscribedFromMarketing: !settingsUpdate.marketing,
+          unsubscribedFromMembersClub: !settingsUpdate.membersClub ?? false,
+          unsubscribedFromAll: false,
+        });
+
+        if (
+          response.updateEmailNotificationSettings?.viewer?.email?.emailNotificationSettings &&
+          settingType in
+            response.updateEmailNotificationSettings.viewer.email.emailNotificationSettings
+        ) {
+          const settings =
+            response.updateEmailNotificationSettings.viewer.email.emailNotificationSettings;
+          if (settings[settingType as keyof EmailSettings] === newSettingValue) {
+            setEmailSettings(settingsUpdate);
+            pushToast({
+              message: `Settings successfully updated. You will ${
+                settingsUpdate[settingType] ? 'no longer' : 'now'
+              } receive ${settingTitle} emails.`,
+            });
+          }
+        }
```

Comment:
you may encounter more typecheck issues here. to fix, simply edit the `settingType` field on `handleToggleProps` below

![image](https://github.com/gallery-so/gallery/assets/12162433/0923e88b-34b4-423b-b902-ac31816dcbb5)

then we should be able to toss this `settingType as keyof EmailSettings` and everything should just work!